modules/vpc-serverless-connector-beta/main.tf

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# Pending new google-cloud-beta provider release Estimated Release 03/22
+# https://github.com/hashicorp/terraform-provider-google/issues/8475
+resource "google_vpc_access_connector" "connector_beta" {
+  for_each = { for connector in var.vpc_connectors : connector.name => connector }
+  provider = google-beta
+  name     = each.value["name"]
+  project  = var.project_id
+  region   = each.value["region"]
+  subnet {
+    name       = each.value["subnet_name"]
+    project_id = lookup(each.value, "host_project_id", null)
```

Comment:
Wouldn't there be an error if the user doesn't provide host_project_id, machine_type  etc as input is a list of objects.
We should follow [this](https://github.com/terraform-google-modules/terraform-google-network/blob/95daff5e79324ebf214c396fe67ce5f997ca0a30/modules/routes/variables.tf#L26) approach and sacrifice some type safety for a better UX

Reply:
Modified to list of maps